clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -740,13 +740,16 @@ Sema::ActOnDecompositionDeclarator(Scope *S, Declarator &D,
     return nullptr;
   }
 
-  Diag(Decomp.getLSquareLoc(),
-       !getLangOpts().CPlusPlus17
-           ? diag::ext_decomp_decl
-           : D.getContext() == DeclaratorContext::Condition
-                 ? diag::ext_decomp_decl_cond
-                 : diag::warn_cxx14_compat_decomp_decl)
-      << Decomp.getSourceRange();
+  unsigned DiagID;
+  if (!getLangOpts().CPlusPlus17)
+    DiagID = diag::ext_decomp_decl;
+  else if (D.getContext() == DeclaratorContext::Condition) {
+    DiagID = getLangOpts().CPlusPlus26 ? diag::warn_cxx26_decomp_decl_cond
+                                       : diag::ext_decomp_decl_cond;
+  } else
+    DiagID = diag::warn_cxx14_compat_decomp_decl;
```

Comment:
Nit: Our code style is against inconsistent bracing
```suggestion
  if (!getLangOpts().CPlusPlus17) {
    DiagID = diag::ext_decomp_decl;
  } else if (D.getContext() == DeclaratorContext::Condition) {
    DiagID = getLangOpts().CPlusPlus26 ? diag::warn_cxx26_decomp_decl_cond
                                       : diag::ext_decomp_decl_cond;
  } else {
    DiagID = diag::warn_cxx14_compat_decomp_decl;
  }
```

Reply:
Or maybe don't brace any branch at all instead.